controller/space.go

```diff
@@ -435,13 +435,24 @@ func deleteOpenShiftResource(
 			}
 		}
 	}
+
 	if len(errorsList) != 0 {
 		var errString string
 		for _, err = range errorsList {
 			errString += fmt.Sprintf("%s\n", err)
 		}
 		return errors.NewInternalErrorFromString(errString)
 	}
+
+	//delete pipelines from the space
+	spaceLabel := space.Data.Attributes.Name
+	log.Debug(ctx, map[string]interface{}{"label": spaceLabel}, "deleting pipelines in")
+	resp, err = cl.DeletePipelines(ctx, client.DeletePipelinesPath(), spaceLabel)
```

Comment:
I must say that I don't like the space controller to be responsible for anything but spaces. This is one good use case where event sourcing can really help us bring more sanity into this.

Reply:
True, this is has been [discussed](https://github.com/openshiftio/openshift.io/issues/3802#issuecomment-404392389) a long time back. However, there is no movement is happening on that side.  So instead of procrastinating the solution, it's better to fix it at best possible place i.e. delete space point, eventually move it to other service.